default.nix

```diff
@@ -24,5 +24,29 @@ if ! builtins ? nixVersion || builtins.compareVersions requiredVersion builtins.
   ''
 
 else
-
-  import ./pkgs/top-level/impure.nix
+{
+  # See ./doc/patching-nixpkgs.xml for documentation
+  patches ? (_pkgs : [])
+, ...
+} @ argsWithPatches:
+let
+  unpatchedArgs = builtins.removeAttrs argsWithPatches [ "patches" ];
+  bootstrapArgs = unpatchedArgs // { overlays = []; crossOverlays = []; };
+
+  unpatchedPkgs = impure unpatchedArgs;
+  bootstrapPkgs = impure bootstrapArgs;
+  patchedPkgs = import patchedNixpkgsDir unpatchedArgs;
```

Comment:
This is using IFD, which seems especially dangerous for the entire Nixpkgs. Not necessarily disqualifying, but we should probably have some idea of what that means.